packages/vm/src/evm/opcodes/util.ts

```diff
@@ -261,3 +261,35 @@ export function updateSstoreGas(
     return new BN(common.param('gasPrices', 'sstoreSet'))
   }
 }
+
+export const eof1CodeAnalysis = (code: Buffer) => {
+  const secCode = 0x01
+  const secData = 0x02
+  const secTerminator = 0x00
+  let codeSize = 0
+  const sectionSizes = {
+    code: 0,
+    data: 0,
+  }
+  if (code[1] === 0x00 && code[2] === 0x01) {
+    if (code.length > 7 && code[3] === secCode && code[6] === secTerminator) {
```

Comment:
What if `code.length` is 7 or less? (Then this EOF format is invalid, right? - it does not throw here?)

Is 7 correct?

Format, magic, version, 0x01, uint16 -> 6 bytes. It is invalid because code is now 0, but it would also not throw here

Reply:
>What if code.length is 7 or less? (Then this EOF format is invalid, right? - it does not throw here?)

You also need the `secTerminator` which is 00 so a compliant header would look like `ef 00 01 01 00 01 00` which would indicate an EOF1 with a one byte code section expected after the section terminator.